src/data/colleges/en.ts

```diff
@@ -111,37 +129,51 @@ const engineeringRequirements: readonly CollegeOrMajorRequirement[] = [
       'https://www.engineering.cornell.edu/students/undergraduate-students/advising/liberal-studies',
     checker: [
       (course: Course): boolean =>
-        engineeringLiberalArtsDistributions.some(
-          distribution => hasCategory(course, distribution) ?? false
+        Object.values(engineeringLiberalArtsGroups).some(group =>
+          group.some(distribution => hasCategory(course, distribution) ?? false)
         ) || courseIsForeignLang(course),
     ],
     fulfilledBy: 'courses',
     perSlotMinCount: [6],
     slotNames: ['Course'],
     additionalRequirements: {
-      'Courses must be from 3 categories.': {
+      'Courses must be from 3 groups.': {
         checker: [
           (course: Course): boolean =>
-            (course.catalogDistr?.includes('LA') || course.catalogDistr?.includes('LAD')) ?? false,
-          ...engineeringLiberalArtsDistributions
-            .filter(it => it !== 'LA' && it !== 'LAD')
-            .map(distribution => (course: Course): boolean =>
-              hasCategory(course, distribution) ?? false
+            engineeringLiberalArtsGroups['Group 1'].some(
```

Comment:
Could you try mapping over the list of groups first so that it's more readable and there's less code duplication 🙏